ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -168,6 +176,18 @@ private ParsePayload(String schemasLocation) {
         throw e;
       }
 
+      if (metadataStore == null) {
+        metadataStore = PipelineMetadataStore.of(schemasLocation, BeamFileInputStream::open);
+      }
+
+      try {
+        deprecationCheck(attributes);
+      } catch (DeprecatedMessageException e) {
+        PerDocTypeCounter.inc(attributes, "error_deprecated_message");
+        PerDocTypeCounter.inc(attributes, "error_submission_bytes", submissionBytes);
```

Comment:
Seeing this here makes me realize that we aren't properly setting these metrics for messages sent to error output via `MessageScrubber.scrub` either. But that problem is not in scope for this PR.